src/System.Buffers.ReaderWriter/System/Buffers/Reader/BufferReader.cs

```diff
@@ -5,25 +5,28 @@
 using System.Diagnostics;
 using System.Runtime.CompilerServices;
 
-namespace System.Buffers.Reader
+namespace System.Buffers
 {
     public ref partial struct BufferReader<T> where T : unmanaged, IEquatable<T>
     {
+        private static NewArray<T> s_newArray = new NewArray<T>();
         private SequencePosition _currentPosition;
         private SequencePosition _nextPosition;
+        private ArrayPool<T> _arrayPool;
         private bool _moreData;
 
         /// <summary>
         /// Create a <see cref="BufferReader" over the given <see cref="ReadOnlySequence{T}"/>./>
         /// </summary>
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
-        public BufferReader(in ReadOnlySequence<T> buffer)
+        public BufferReader(in ReadOnlySequence<T> buffer, ArrayPool<T> arrayPool = null)
```

Comment:
What's our current thinking on `ArrayPool<T>`?  Do we expect developers using this API will actually be creating their own pools?  I'm wondering if we should instead just use Shared, and only add such a ctor if there's real demand.

Reply:
Yes, we expect them to create their own wrappers **if** this matters to them. We can't use the regular array pool as there is no reasonable way to manage lifetime when you don't know if the Span you got back is rented or not.

I'll be creating some concrete examples of how to do this and will be updating this pull with them (and measurements).